tests/unit/components/ModelizerTextView.spec.js

```diff
@@ -1,30 +1,227 @@
+  let emit;
 
   beforeEach(() => {
+    openFileSubscribe = jest.fn();
+    openFileUnsubscribe = jest.fn();
+    updateRemoteSubscribe = jest.fn();
+    updateRemoteUnsubscribe = jest.fn();
+    checkoutSubscribe = jest.fn();
+    checkoutUnsubscribe = jest.fn();
+
+    GitEvent.UpdateRemoteEvent.subscribe.mockImplementation(() => {
+      updateRemoteSubscribe();
+      return { unsubscribe: updateRemoteUnsubscribe };
+    });
+    FileEvent.OpenFileEvent.subscribe.mockImplementation(() => {
+      openFileSubscribe();
+      return { unsubscribe: openFileUnsubscribe };
+    });
+    GitEvent.CheckoutEvent.subscribe.mockImplementation(() => {
+      checkoutSubscribe();
+      return { unsubscribe: checkoutUnsubscribe };
+    });
+
     wrapper = shallowMount(ModelizerTextView, {
       props: {
-        viewType: 'model',
+        viewType: 'text',
+        projectName: 'project-00000000',
       },
-      stubs: {
-        GitBranchCard: true,
+      mocks: {
+        FileEvent,
+        GitEvent,
       },
     });
   });
 
   describe('Test variables initialization', () => {
     describe('Test props: viewType', () => {
-      it('should match "model"', () => {
-        expect(wrapper.vm.viewType).toEqual('model');
+      it('should match "text"', () => {
+        expect(wrapper.vm.viewType).toEqual('text');
+      });
+    });
+
+    describe('Test props: projectName', () => {
+      it('should match "project-00000000"', () => {
+        expect(wrapper.vm.projectName).toEqual('project-00000000');
+      });
+    });
+  });
+
+  describe('Test function: onOpenFileEvent', () => {
+    it('should set activeFile value equal to onOpenFileEvent param.id ', () => {
+      wrapper.vm.onOpenFileEvent({ id: 'terraform/app.tf', label: 'app.tf', content: '' });
+
+      expect(wrapper.vm.activeFile).toEqual('terraform/app.tf');
+    });
+
+    it('should push the new file to files Array if it is not already there', () => {
+      wrapper.vm.files = [{ id: 'terraform/app.tf', label: 'app.tf', content: '' }];
+      wrapper.vm.onOpenFileEvent({ id: 'README.md', label: 'README.md', content: '' });
+
+      expect(wrapper.vm.files).toEqual([{ id: 'terraform/app.tf', label: 'app.tf', content: '' }, { id: 'README.md', label: 'README.md', content: '' }]);
+    });
+
+    it('should not push the new file to files Array if it is already there', () => {
+      wrapper.vm.files = [{ id: 'terraform/app.tf', label: 'app.tf', content: '' }];
+      wrapper.vm.onOpenFileEvent({ id: 'terraform/app.tf', label: 'app.tf', content: '' });
+
+      expect(wrapper.vm.files).toEqual([{ id: 'terraform/app.tf', label: 'app.tf', content: '' }]);
+    });
+  });
+
+  describe('Test function: closeFile', () => {
+    beforeEach(() => {
+      emit = jest.fn();
+
+      FileEvent.SelectedFileEvent.next.mockImplementation(() => emit());
+    });
+
+    describe('The file to close is contained within files Array', () => {
+      describe('files length is 2 or more', () => {
```

Comment:
too much `describe`

Reply:
I wanted to use `context` but it was not possible, so I used two `describe` to bring clarity and more context to the tests (the function tested has lots of conditions)
So do you prefer I just use one big `describe` instead of two ?
@k-y0u @Zorin95670

Reply:
Considering that your `describe` in `describe` contains only when `it`, you could only merge the details of the `describe` in the `it`.

Reply:
done